src/main/java/org/minions/devfund/damianvp/moviedatabase/MovieDatabase.java

```diff
@@ -0,0 +1,150 @@
+    private ArrayList<Actor> actorList;
+
+    /**
+     * Constructor class.
+     */
+    MovieDatabase() {
+        this.movieList = new ArrayList<>();
+        this.actorList = new ArrayList<>();
+    }
+
+    /**
+     * Method to get the movie list.
+     * @return ArrayList<Movie>, movieList.
+     */
+    ArrayList<Movie> getMovieList() {
+        return this.movieList;
+    }
+
+    /**
+     * Method to get the actor list.
+     * @return ArrayList<Actor>, actorList.
+     */
+    ArrayList<Actor> getActorList() {
+        return this.actorList;
+    }
+
+    /**
+     * Method to add a Movie.
+     * @param name String type.
+     * @param actors String List type
+     */
+    void addMovie(final String name, final String[] actors) {
+        if (!containsMovie(name)) {
+            Movie movie = getMovieObject(name);
+            for (int i = 0; i < actors.length; i++) {
+                if (containsActor(actors[i])) {
+                    for (int j = 0; j < this.actorList.size(); j++) {
+                        if (this.actorList.get(j).getName().equals(actors[i])) {
+                            this.actorList.get(j).addMovie(movie);
+                            movie.addActor(this.actorList.get(j));
+                        }
+                    }
+
+
+                } else {
+                    Actor actor = getActorObject(actors[i]);
+                    actor.addMovie(movie);
+                    movie.addActor(actor);
+                    this.actorList.add(actor);
+                }
+            }
+            this.movieList.add(movie);
+        }
+    }
+
+    /**
+     * Method to check if movie already exist in class.
+     * @param name String type.
+     * @return true: if movie exist.
+     *         false: if movie does not exist.
+     */
+    public boolean containsMovie(final String name) {
+        return this.movieList.stream().anyMatch(o -> o.getName().equals(name));
+    }
+
+    /**
+     * Method to check if actor already exist in class.
+     * @param name String type.
+     * @return true: if actor exist.
+     *         false: if actor does not exist.
+     */
+    public boolean containsActor(final String name) {
+        return this.actorList.stream().anyMatch(o -> o.getName().equals(name));
+    }
+
+    /**
+     * Method to get the Actor object.
+     * @param name String type.
+     * @return Actor object.
+     */
+    public Actor getActorObject(final String name) {
+        return this.actorList.stream().findAny().filter(o -> o.getName().equals(name)).orElse(new Actor(name));
+    }
+
+    /**
+     * Method to get the Movie object form the list by movie name.
+     * @param name String type.
+     * @return Movie object from list: if movie already exists.
+     *         new Movie object: if movie does not exists.
+     */
+    public Movie getMovieObject(final String name) {
+        Movie ob = this.movieList.stream().findAny().filter(o -> o.getName().equals(name)).orElse(new Movie(name));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "ob". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)